python/ray/serve/backend_worker.py

```diff
@@ -195,8 +196,18 @@ async def invoke_batch(self, request_item_list):
             self.request_counter.add(batch_size)
             result_list = await call_method(*arg_list, **kwargs_list)
 
-            if (not isinstance(result_list,
-                               list)) or (len(result_list) != batch_size):
+            if not isinstance(result_list, Iterable) or isinstance(
+                    result_list, (dict, set)):
+                error_message = ("Worker returned a {}. RayServe expect "
```

Comment:
usually clearer to reverse the message - "RayServe expects ... but worker returned a ..."